### PR TITLE
fix(ui): ensure map type variables can get selected key

### LINKED ISSUE #13749: Map variable type doesn't work
to repro, try to create a map variable type and use it in a dashboard. you cannot select any value.

Can we create an e2e test for this?

Kapture 2019-05-01 at 23 08 11

### PR DESCRIPTION
**Pull request by ischolten**
Closes #13749

Describe your proposed changes here.

- [x] [Well-formatted commit messages](https://www.conventionalcommits.org/en/v1.0.0-beta.3/)
- [x] Rebased/mergeable
- [x] Tests pass

### CODE REVIEW

File: ui/src/dashboards/selectors/index.ts [hunk 1]
```diff
@@ -58,7 +58,7 @@ export const getVariableValuesForDropdown = (
       const selection = list.find(({value}) => value === selectedValue)
 
       return {
-        selectedKey: selection[0],
+        selectedKey: selection.name,
```
Comment (OfTheDelmer) [inline ui/src/dashboards/selectors/index.ts]: wooops!

File: ui/src/dashboards/components/variablesControlBar/VariableDropdown.test.tsx [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+            },
+          },
+          order: ['03cbdc8a53a63000'],
+        },
+      },
+    },
+  }
+}
+
+describe('Dashboards.Components.VariablesControlBar.VariableDropdown', () => {
```
Comment (alexpaxton) [inline ui/src/dashboards/components/variablesControlBar/VariableDropdown.test.tsx]: Yay more tests

File: ui/src/dashboards/selectors/index.ts (hunk 1 above)
Comment (alexpaxton) [inline ui/src/dashboards/selectors/index.ts]: Does this need a guard?

Comment (ischolten) [reply ui/src/dashboards/selectors/index.ts]: i can add a _.get

**Pull request merged**